recipes/spirv-cross/all/conanfile.py

```diff
@@ -0,0 +1,175 @@
+import glob
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class SpirvCrossConan(ConanFile):
+    name = "spirv-cross"
+    description = "SPIRV-Cross is a practical tool and library for performing " \
+                  "reflection on SPIR-V and disassembling SPIR-V back to high level languages."
+    license = "Apache-2.0"
+    topics = ("conan", "spirv-cross", "reflection", "disassembler", "spirv", "spir-v", "glsl", "hlsl")
+    homepage = "https://github.com/KhronosGroup/SPIRV-Cross"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_executable": [True, False],
+        "glsl": [True, False],
+        "hlsl": [True, False],
+        "msl": [True, False],
+        "cpp": [True, False],
+        "reflect": [True, False],
+        "c_api": [True, False],
+        "util": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_executable": True,
+        "glsl": True,
+        "hlsl": True,
+        "msl": True,
+        "cpp": True,
+        "reflect": True,
+        "c_api": True,
+        "util": True
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if not self.options.glsl and \
+           (self.options.hlsl or self.options.msl or self.options.cpp or self.options.reflect):
+            raise ConanInvalidConfiguration("hlsl, msl, cpp and reflect require glsl enabled")
+        if self.options.shared:
+            # these options don't contribute to shared binary
+            del self.options.c_api
```

Comment:
how come C API doesn't contribute to shared library?

Reply:
Because source files of C APi are always added to spirv-cross if shared.

```cmake
if (SPIRV_CROSS_SHARED)
	set(SPIRV_CROSS_VERSION ${spirv-cross-abi-major}.${spirv-cross-abi-minor}.${spirv-cross-abi-patch})

	if (NOT SPIRV_CROSS_SKIP_INSTALL)
		configure_file(
			${CMAKE_CURRENT_SOURCE_DIR}/pkg-config/spirv-cross-c-shared.pc.in
			${CMAKE_CURRENT_BINARY_DIR}/spirv-cross-c-shared.pc @ONLY)
		install(FILES ${CMAKE_CURRENT_BINARY_DIR}/spirv-cross-c-shared.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
	endif()

	spirv_cross_add_library(spirv-cross-c-shared spirv_cross_c_shared SHARED
			${spirv-cross-core-sources}
			${spirv-cross-c-sources})
```

Reply:
Maybe it makes more sense for consumers to keep `c_api` is shared and raise `ConanInvalidConfiguration` if `shared=True` and `c_api=False`.
(util option would still be deleted because it is not included in shared build).

Reply:
To give you more insight:
- if shared, spirv-cross only install C API. Even though, at compile time glsl, hlsl etc can be enabled, but it doesn't change the C API.
- if static, C API can be installed, but also finer grain C++ APIs.

Consequence is that some libs like shaderc which optionnaly depends on spirv-cross, can't use spirv-cross shared because they expect C++ API.